### PR TITLE
feat: implement in-memory queue and storage

### LINKED ISSUE #72: Implement in-memory queue and storage

### PR DESCRIPTION
**Pull request by sd1li**
Closes #72 

`StorageBackend.Download`: `io.Writer` was changed to return an `io.ReadCloser` as writer didn't look correct.

### CODE REVIEW

**Comment by sd1li** [comment]
@poundifdef I added them to a single package so the selector is `memory.{Queue,Storage}` which reads nicer than something like `memoryqueue.Queue` or `memorystorage.Storage`

File: pkg/memory/queue.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package memory
+
+import (
+	"fmt"
+	"scratchdata/pkg/queue"
+	"sync"
+)
+
+var (
+	_ queue.QueueBackend = (*Queue)(nil)
```
Comment (poundifdef) [inline pkg/memory/queue.go:10]: What does this do?

Comment (sd1li) [reply pkg/memory/queue.go:10]: It makes compilation fail if the `queue.QueueBackend` interface changes and `Queue` doesn't implement the changes.

Comment (poundifdef) [reply pkg/memory/queue.go:10]: Okay. When we actually start using these objects in `main.go` this will no longer be needed. (We can delete it then.)

**Comment by sd1li** [comment]
> Please use this package structure instead:
> 
>     * `pkg/filestore/memory/memory.go`
> 
>     * `pkg/queue/memory/memory.go`

Done.

**Comment by poundifdef** [comment]
Thank you!

**Pull request merged**